app/scripts/formValidation.js

```diff
@@ -3,6 +3,44 @@ const fullnameErr = document.querySelector('#fullname-err');
 const emailErr = document.querySelector('#email-err');
 const msgErr = document.querySelector('#msg-err');
 const successMsg = document.querySelector('#success-message');
+const fullnameField = document.getElementById('fullname');
+const emailField = document.getElementById('email');
+const messageField = document.getElementById('message');
+const info = {
+  fullname: '',
+  email: '',
+  message: '',
+};
+
+window.onload = () => {
+  const data = localStorage.getItem('userData');
+  if (data !== null) {
+    const parsedData = JSON.parse(data);
+    fullnameField.value = parsedData.fullname;
+    emailField.value = parsedData.email;
+    messageField.value = parsedData.message;
+  }
+};
+
+const storeData = (info) => {
+  const convert = JSON.stringify(info);
+  localStorage.setItem('userData', convert);
+};
+
+emailField.addEventListener('change', (e) => {
+  info.email = e.target.value;
+  storeData(info);
+});
+
+fullnameField.addEventListener('change', (e) => {
+  info.fullname = e.target.value;
+  storeData(info);
+});
+
+messageField.addEventListener('change', (e) => {
+  info.message = e.target.value;
+  storeData(info);
+});
```

Comment:
### 👏🏾  Well done implementing this data storage feature. But I notice a slight issue that will improve your portfolio. 

To see the issue follow the steps below 👇🏾 

- **_Step 1_**: Type data into `all three` input fields - `Full name`, `email`, and `message` at first. 
- **_Step 2_**: Refresh the page, the saved input data is loaded on `all` the in1: put fields as expected. 
- **_Step 3_**: If you now make a change to the `email` input fields and then reload the browser, the other input fields (full name and message) data stored is cleared. 

This is unexpected behavior. I trace this fault to the below code 👇🏾 

```javascript
emailField.addEventListener('change', (e) => {
  info.email = e.target.value;
  storeData(info);
});

fullnameField.addEventListener('change', (e) => {
  info.fullname = e.target.value;
  storeData(info);
});

messageField.addEventListener('change', (e) => {
  info.message = e.target.value;
  storeData(info);
});
```
#### **When the email field is changed**
When the value of the email input changes, the callback function is executed, updating the info object with the new email value and then potentially storing the data using the `storeData` function. At this point the `info` data object passed into the  `storeData` function now looks like this 👇🏾 

```javascript
const info = {
  fullname: '',
  email: 'abc@email.com',
  message: '',
};
```

You can observe that the value of `full name` and `message` of the info data object are empty strings being passed into the `storeData(info)`, thereby updating the localStorage userData with the empty string for `full name` and `message`. On page reload, an empty string is retrieved and the input fields show empty data instead of the previously stored data from **_step2_**.

To fix this, use the code below 👇🏾 

```javascript
const userData = localStorage.getItem('userData');
if (userData) {
  try {
    const parsedData = JSON.parse(userData);
    Object.assign(info, parsedData);
  } catch (error) {
    console.error('Error parsing userData from local storage:', error);
  }
}
```

In this code, localStorage.getItem('userData') retrieves the value associated with the key "userData" from the local storage. If the value exists (userData is truthy), the code attempts to parse it as JSON using JSON.parse(). The parsed data is then merged into the info object using Object.assign().

Please note that JSON.parse() expects the stored data to be in valid JSON format. If the stored data is not valid JSON, an error will be thrown. In such cases, the error will be logged to the console, but the code execution will continue.

After running this code, the info object will contain the loaded data from local storage if "userData" exists, or it will be initialized with empty values otherwise.